2.1/slidingWindow_test.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+	"testing"
+)
+
+//test setup
+var (
+	seed     int64 = 1
+	testRand       = rand.New(rand.NewSource(seed))
+)
+
+func TestRecordValue(t *testing.T) {
+	r := &R{data: seedData([]uint{0, 1, 2, 3}), percentile: .95, window: 100}
+	expectedLength := len(r.data) + 1
+	r.RecordValue(101)
+	if len(r.data) != expectedLength {
+		t.Errorf("RecordValue: expected length: %v, actual: %v", expectedLength, len(r.data))
+	}
+
+}
+
+func TestGetPercentile(t *testing.T) {
+	var expected = []uint{0, 48, 98, 148, 198, 248}
+	var result uint
+	testData := createData(testRand, 300)
+	r := &R{data: seedData([]uint{}), percentile: .95, window: 50}
```

Comment:
Could you potentially change this to a even integer percentile or a window of 100?  It isn't instantly apparent what the 95th percentile of 1-50 is.  It is much easier to calculate the 90th (i = 45) or 80th (i = 40) of a window of 50, or any percentile of a window of 100.